src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -183,6 +186,62 @@ def _parse_frontend_config(config) -> Tuple[int, bool]:
     raise LookupError('Failed to determine RGW port from "{}"'.format(config))
 
 
+def radosgw_admin(args: List[str]) -> Tuple[int, str, str]:
+    try:
+        result = subprocess.run(
+            [
+                'radosgw-admin',
+            '-c', str(mgr.get_ceph_conf_path()),
+                '-k', str(mgr.get_ceph_option('keyring')),
+                '-n', f'mgr.{mgr.get_mgr_id()}',
+            ] + args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            timeout=10,
+        )
+        return result.returncode, result.stdout.decode('utf-8'), result.stderr.decode('utf-8')
+    except subprocess.CalledProcessError as ex:
+        mgr.log.error(f'Error executing radosgw-admin {ex.cmd}: {ex.output}')
+        raise
+    except subprocess.TimeoutExpired as ex:
+        mgr.log.error(f'Timeout (10s) executing radosgw-admin {ex.cmd}')
+        raise
+
+
+def configure_rgw_users():
+    mgr.log.info('Configuring dashboard RGW credentials')
+
+    def get_secrets(user: str, out: str) -> Tuple[Optional[str], Optional[str]]:
+        r = json.loads(out)
+        for k in r.get('keys', []):
+            if k.get('user') == user and r.get('system') in ['true', True]:
+                access_key = k.get('access_key')
+                secret_key = k.get('secret_key')
+                return access_key, secret_key
+        return None, None
+
+    # set up dashboard creds
+    user = 'dashboard'
+    access_key = None
+    secret_key = None
+    rc, out, _ = radosgw_admin(['user', 'info', '--uid', user])
```

Comment:
@liewegas 
- We need to optionally pass the `--rgw-realm` param as users are per realm in case of multisite.
  In case of multisite the logic for setting up credentials per real is required, similar to:
  https://github.com/ceph/ceph/pull/41590/files#diff-95f10b8f6c554f23d766c56a7549b7f05291ff95e6d96a106e6980e49439d62eR1006
- In Dashboard we haven't yet adapted the code (backend & frontend) to set up credentials per realm instead of per daemon. @epuertat Should this happen in a follow-up PR?

Reply:
I've updated this PR to handle the multisite case (generate a dictionary of realm to key as the value).  Not sure about the dashboard side of things, though!